src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/UpToDate/BuildUpToDateCheck.cs

```diff
@@ -676,39 +688,58 @@ private bool CheckCopyToOutputDirectoryFiles(Log log, in TimestampCache timestam
 
                     filename = _configuredProject.UnconfiguredProject.MakeRelative(filename);
 
-                    log.Info(nameof(Resources.FUTD_CheckingPreserveNewestFile_1), rootedPath);
+                    log.Info(nameof(Resources.FUTD_CheckingCopyToOutputDirectoryItem_3), itemType, item.CopyType.ToString(), sourcePath);
+                    log.Indent++;
 
-                    DateTime? itemTime = timestampCache.GetTimestampUtc(rootedPath);
+                    DateTime? sourceTime = timestampCache.GetTimestampUtc(sourcePath);
 
-                    if (itemTime != null)
+                    if (sourceTime != null)
                     {
-                        log.Indent++;
-                        log.Info(nameof(Resources.FUTD_SourceFileTimeAndPath_2), itemTime, rootedPath);
-                        log.Indent--;
+                        log.Info(nameof(Resources.FUTD_SourceFileTimeAndPath_2), sourceTime, sourcePath);
                     }
                     else
                     {
-                        return log.Fail("CopyToOutputDirectorySourceNotFound", nameof(Resources.FUTD_CheckingPreserveNewestFileSourceNotFound_1), rootedPath);
+                        return log.Fail("CopyToOutputDirectorySourceNotFound", nameof(Resources.FUTD_CheckingCopyToOutputDirectorySourceNotFound_1), sourcePath);
                     }
 
-                    string destination = Path.Combine(outputFullPath, filename);
-                    DateTime? destinationTime = timestampCache.GetTimestampUtc(destination);
+                    string destinationPath = Path.Combine(outputFullPath, filename);
+                    DateTime? destinationTime = timestampCache.GetTimestampUtc(destinationPath);
 
                     if (destinationTime != null)
                     {
-                        log.Indent++;
-                        log.Info(nameof(Resources.FUTD_DestinationFileTimeAndPath_2), destinationTime, destination);
-                        log.Indent--;
+                        log.Info(nameof(Resources.FUTD_DestinationFileTimeAndPath_2), destinationTime, destinationPath);
                     }
                     else
                     {
-                        return log.Fail("CopyToOutputDirectoryDestinationNotFound", nameof(Resources.FUTD_CheckingPreserveNewestFileDestinationNotFound_1), destination);
+                        return log.Fail("CopyToOutputDirectoryDestinationNotFound", nameof(Resources.FUTD_CheckingCopyToOutputDirectoryItemDestinationNotFound_1), destinationPath);
                     }
 
-                    if (destinationTime < itemTime)
+                    if (item.CopyType == CopyType.PreserveNewest)
+                    {
+                        if (destinationTime < sourceTime)
```

Comment:
The non-obvious fact here is that when `log.Fail` is called, the indent value is reset to zero. The only way to exit this block is by normal iteration (which increments/decrements correctly) or by `return log.Fail` which sets it back to zero so that we always show failures in column zero.